gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergTable.java

```diff
@@ -213,7 +215,10 @@ public DatasetDescriptor getDatasetDescriptor(FileSystem fs) {
   protected void registerIcebergTable(TableMetadata srcMetadata, TableMetadata dstMetadata) {
     if (dstMetadata != null) {
       // use current destination metadata as 'base metadata' and source as 'updated metadata' while committing
-      this.tableOps.commit(dstMetadata, srcMetadata.replaceProperties(dstMetadata.properties()));
+      Map<String, String> combinedMetadataProperties = Maps.newHashMap();
+      combinedMetadataProperties.putAll(dstMetadata.properties());
+      combinedMetadataProperties.putAll(srcMetadata.properties());
```

Comment:
please add a comment that captures the reasoning behind overshadowing dest with src